packages/perseus/src/widgets/categorizer/prompt-utils.ts

```diff
@@ -0,0 +1,34 @@
+import {WidgetType} from "../../prompt-types";
+
+import type categorizer from "./categorizer";
+import type {PerseusCategorizerUserInput} from "../../validation.types";
+import type {PropsFor} from "@khanacademy/wonder-blocks-core";
+
+type WidgetProps = PropsFor<typeof categorizer.widget>;
+
+export type CategorizerPromptJSON = {
+    type: WidgetType;
+    options: {
+        items: WidgetProps["items"];
```

Comment:
Any idea why these prop types come through as optional? They all have `| undefined` as part of the type, even though the props get [defined from this type](https://github.com/Khan/perseus/blob/main/packages/perseus/src/perseus-types.ts#L331), which isn't optional.

Reply:
Hrm. Well, maybe `PropsFor<>` isn't the right call.  They are optional here because the `Categorizer` defines a default prop for `items`, meaning `PropsFor<>` calculates that `items` can be `undefined` because it's got a default. 

![Uploading image.png…]()

Reply:
Do you think using `React.ComponentProps<typeof categorizer.widget>;` would be a suitable alternative?

Reply:
That's a good thought Matthew. Typescript will still ensure the types compatible even if we define them manually. Given that `items` is just a `ReadonlyArray<string>` I think it's a good idea to not reference the Prop types in these LLM-specific types. 

@MikeKlemarewski would you mind adjusting `items` and `categories` to be defined manually? (ie. `{items: ReadonlyArray<string>, categories: ReadonlyArray<string>}`)